### ISSUE TITLE
Remove freetype and make truetype the default, maybe.

### ISSUE THREAD

**Issue by robert-strandh** [robert-strandh/McCLIM#20]
I am contemplating two improvements:

1. Make mcclim-truetype the default when mcclim is built using ASDF.
   I have used it enough that I think it is working fine.  It uses the
   XRender extension, so it is entirely compatible with the normal CLX
   backend.  I suppose I would also rename the current mcclim system
   so that it would still be available for those who don't want
   truetype fonts.

2. Remove the system mcclim-freetype which serves the same purpose as
   mcclim-truetype, but which uses FFI.

I am not going to do it immediately, because I want to wait for
reactions from the current users.  So let me know what you think.

**Comment by gabriel-laddel** [comment]
Both sound AOK to me. Freetype, at least afaik is a distraction, and cutting down on the line count is always a good thing. The (unpublished, largely unchanged) branch I'm working on is down to 104,702 lines of lisp & 71 lines of sh. Some clever macros and cutting out the goatee text substrate (which drei supersedes?) might be able to get it to sub 100k.

That said, it seems like show stopping errors occur more often when using the whole CLIM desktop (listener, debugger climacs) & truetype than when not This can be avoided by using the swank debugger (informal error report: all CLIM applications freeze, refuse to repaint and an error about reaching out of bounds in an array is thrown in the swank debugger, even if not enabled. A 'restart the CLX command loop' is available, but does nothing).

**Comment by gas2serra** [comment]
It's ok to me.

**Comment by robert-strandh** [comment]
I removed the Freetype code, but I have not yet made Truetype the
default.  I don't have a problem with the Truetype code itself.
Rather, I hesitate modifying the ASDF system definitions so that
building MCCLIM automatically builds MCCLIM-TRUETYPE as well.  In
fact, I don't know at this point whether changing the system
definitions is the right thing to do.  It might be the case that the
CLX backend should be changed, because that seems to be what
MCCLIM-TRUETYPE does.

I will continue to investigate the best way of doing this, but it will
not happen immediately.

**Comment by gabriel-laddel** [comment]
Regarding (one of) the showstopping errors mentioned previously: Eval (truetype loaded), ```(climi::with-text-size (t 1200) (princ "?"))``` at the listener, which will throw http://paste.lisp.org/display/311273 

If you attempt to RESTART-EVENT-LOOP, it fails, and you may have to kill all threads associated with any running CLIM applications to dismiss them. If you utilize CONTINUE over and over again,

"Asynchronous REQUEST-ERROR in request 701 (last request was 702)  Code 0.0 [error]
   [Condition of type XLIB:REQUEST-ERROR]"

"Asynchronous REQUEST-ERROR in current request  Code 0.0 [error]
   [Condition of type XLIB:REQUEST-ERROR]"

"Asynchronous REQUEST-ERROR in request 703 (last request was 702)  Code 0.0 [error]
   [Condition of type XLIB:REQUEST-ERROR]"

...

"Asynchronous REQUEST-ERROR in request 721 (last request was 702)  Code 0.0 [error]
   [Condition of type XLIB:REQUEST-ERROR]"

And finally we get disconnected.

"couldn't read from #<SB-SYS:FD-STREAM
                     for "socket, peer: /tmp/.X11-unix/X0"
                     {101CC370C3}>:
  Connection reset by peer
   [Condition of type SB-INT:SIMPLE-STREAM-ERROR]"

Then invoke "2: [ABORT] Abort thread (#<THREAD "Listener" RUNNING {101C33E443}>)". There will be another sldb error for CLX. You can abort that too. If you attempt to launch a CLIM application from this point, it will fail due to a broken pipe. Therp & Fare posted some code about this once upon a time ```(clim:destroy-port (car climi::*all-ports*))``` Run it now and try relaunching. Success!

The nature of the errors leads me to believe that something has gone wrong on the X side in the C code (duh). If someone knows of a way to interrogate or instrument an X process, I'm all ears. This would probably be pretty useful for CLIM in general.

If you have some objects that were presented prior to the restart, and you attempt to display them again on the listener, you'll get an error ~"dead X display connection". Recompile / regenerate the objects and it'll work fine.

**Comment by dkochmanski** [comment]
I have encountered serious performance issues with truetype on CCL (probably other compilers have them too). I've also checked, if using the removed freetype implementation helps – it doesn't. In PR #26 truetype is default for SBCL, but needs to be loaded separately for basically everything else.

As already said, freetype implementation is as slow as truetype, so no need to keep it. We need to find the bottleneck on other implementations, not sure where it is. I suspect it's somewhere in CLX (?).
